# NumPy VS Python with Large Arrays

## NumPy

In [5]:
import numpy as np

In [36]:
%%timeit
rng = np.random.default_rng(seed=1)
bigarray = rng.integers(low=1, high=100, size=(1000000, 10))
bigarray %= 3
bigarray

42.8 ms ± 238 μs per loop (mean ± std. dev. of 7 runs, 10 loops each)


## Python

In [37]:
%%timeit
bigarray = rng.integers(low=1, high=100, size=(1000000, 10))
for n in bigarray:
    n %= 3
bigarray

491 ms ± 2.02 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)


## Results

> NumPy's time was 42.8 ms ± 238 μs per loop
> Python's time was 491 ms ± 2.02 ms per loop

NumPy's time per loop was 11.47 times more efficient than Python's using modulo.